types.cc

```diff
@@ -2876,6 +2876,16 @@ bool abstract_type::is_value_compatible_with(const abstract_type& other) const {
     return is_value_compatible_with_internal(*this, *other.underlying_type());
 }
 
+std::optional<size_t>
+user_type_impl::idx_of_field(const bytes& name) const {
+    for (size_t i = 0; i < _field_names.size(); ++i) {
```

Comment:
Would it make sense to use std::find here?

Reply:
This was copied from a non-general implementation, so it makes sense to keep it like this (and maybe make it prettier in a follow up). Code movement + refactoring in the same patch is a recipe for a stealth bug.

Reply:
Sure. Can be in another commit. Just pointing out it looks very much like a standard algorithm.